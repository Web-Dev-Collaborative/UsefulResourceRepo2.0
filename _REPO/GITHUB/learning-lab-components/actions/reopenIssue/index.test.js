const reopenIssue = require('./')
const mockContext = require('../../tests/mockContext')

describe('reopenIssue', () => {
  let context

  beforeEach(() => {
    context = mockContext({
      issue: {
        number: 1
      }
    }, {
      issues: {
        update: jest.fn()
      },
      search: {
        issues: jest.fn(() => Promise.resolve({
          data: {
            items: [{ title: 'My issue', number: 2 }]
          }
        }))
      }
    })
  })

  it('reopens the issue from the webhook payload', async () => {
    await reopenIssue(context, {})
    expect(context.github.issues.update).toHaveBeenCalled()
    expect(context.github.issues.update.mock.calls).toMatchSnapshot()
  })

  it('reopens an issue with an issue string', async () => {
    await reopenIssue(context, { issue: 'My issue' })
    expect(context.github.issues.update).toHaveBeenCalled()
    expect(context.github.issues.update.mock.calls).toMatchSnapshot()
  })

  it('reopens an issue with an issue number', async () => {
    await reopenIssue(context, { issue: 2 })
    expect(context.github.issues.update).toHaveBeenCalled()
    expect(context.github.issues.update.mock.calls).toMatchSnapshot()
  })

  it('throws an error if the issue cannot be found', async () => {
    try {
      await reopenIssue(context, { issue: 'My issue' })
    } catch (err) {
      expect(err).toMatchSnapshot()
    }
  })
})
