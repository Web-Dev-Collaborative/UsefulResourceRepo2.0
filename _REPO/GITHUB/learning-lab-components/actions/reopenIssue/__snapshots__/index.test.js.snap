// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reopenIssue reopens an issue with an issue number 1`] = `
Array [
  Array [
    Object {
      "number": 2,
      "owner": "JasonEtco",
      "repo": "example",
      "state": "open",
    },
  ],
]
`;

exports[`reopenIssue reopens an issue with an issue string 1`] = `
Array [
  Array [
    Object {
      "number": 2,
      "owner": "JasonEtco",
      "repo": "example",
      "state": "open",
    },
  ],
]
`;

exports[`reopenIssue reopens the issue from the webhook payload 1`] = `
Array [
  Array [
    Object {
      "number": 1,
      "owner": "JasonEtco",
      "repo": "example",
      "state": "open",
    },
  ],
]
`;
